### DISCUSSION TITLE
Templates Issue with useSearchParams and Nextjs 14.1.4

### DISCUSSION THREAD

**Discussion by Asvaldrz** [primefaces/community#1519] [PrimeReact]
Hi, I'm using the avalon and sakai templates and trying to use the latest nextjs (14.1.4).
Whats the recommended work around to get around the issues of the useSearchParams causing this error on build?
` ⨯ useSearchParams() should be wrapped in a suspense boundary at page "/". Read more: https://nextjs.org/docs/messages/missing-suspense-with-csr-bailout`

The issue is because useSearchParams is used to detect changes in the layout core file. Which is then getting applied to everything i believe.
https://github.com/primefaces/sakai-react/blob/master/layout/layout.tsx

Sure you can put in the experimental error ignore but is there a better way we can patch the template to be able to follow the recommendation of using the suspense boundaries around it?

When you do turn the experimental error ignore on. What it then does is deopt everything into client side rendering and you get this
`https://nextjs.org/docs/messages/deopted-into-client-rendering`
Which again suggest a suspense boundary.

sakai template
https://github.com/primefaces/sakai-react

Thanks

**Comment by chrishj59** [comment]
I have found if you use the React <Suspense> in layout this has resolved the issue for me.

`<Suspense>
			<Layout>{children}</Layout>
		</Suspense>`

**Reply by Asvaldrz** [reply]
Thanks Chris. 
You found any side effects using this?
I'm wondering if it might cause issues with the components below (but i'm still learning) 
Could it cause child things to lose state? Or issues with them all together passing through that suspense?
